Src/Support/Google.Apis.Core/Requests/Parameters/ParameterValidator.cs

```diff
@@ -29,22 +29,42 @@ public static class ParameterValidator
     {
         /// <summary>Validates a parameter value against the methods regex.</summary>
         [VisibleForTestOnly]
-        public static bool ValidateRegex(IParameter param, string paramValue)
+        public static bool ValidateRegex(IParameter param, string paramValue, out string error)
```

Comment:
This is a breaking change - it's very unfortunate that the method is public at all (instead of internal) but the VisibleForTestOnly attribute doesn't mean we can break it.

I would suggest:

- Adding a new *private* method with the out parameter
- Calling the new method from both this and ValidateParameter below

For ValidateParameter, you'll need to add a new *public* overload because it's called from ClientServiceRequest.

It would be reasonable to deprecate the old methods too.